src/fields/field_values.hh

```diff
@@ -477,6 +536,23 @@ public:
     bool equal_to(const  return_type &other) {
         return arma::max(arma::abs(value_ - other)) < 4*std::numeric_limits<ET>::epsilon();
     }
+    // Multiplied value_ by double coefficient
+    void scale(double scale_coef) {
+        if (is_scalable())
+	        for(unsigned int i=0; i< n_rows(); i++) {
+	            value_.at(i) = scale_coef * value_.at(i);
+	        }
+    }
+
+    // Compute the weighted average of val_0 and val_1
+    void interpolated(double coef, return_type val_0, return_type val_1) {
```

Comment:
Just one implementation of interpolate using double loop (as for a matrix). Move it into TableFunction.